vm/address/src/lib.rs

```diff
@@ -101,10 +105,23 @@ impl Address {
     pub fn protocol(&self) -> Protocol {
         self.protocol
     }
+
     /// Returns data payload of Address
     pub fn payload(&self) -> &[u8] {
         &self.payload
     }
+
+    /// Returns network configuration of Address
+    pub fn network(&self) -> Network {
+        self.network
+    }
+
+    /// Sets the network for the address and returns a mutable reference to it
+    pub fn with_network(&mut self, network: Network) -> &mut Self {
```

Comment:
As per your question, I think it makes more sense to name this `set_network` given its functionality. Just my opinion tho

Reply:
ya, I didn't like that the name didn't indicate self is mutated, with `set_network` it doesn't indicate that it can be chained and returns a mutable self ref. I'll think about this later and see if any examples in std lib for naming

Reply:
with kinda feels like its a temporary change. set feels more permanent

Reply:
yeah I agree this is better, I didn't like that it didn't indicate that it could be chained, but that is definitely less of a problem than it being worded as not being mutated and instead being copied